### PR TITLE
chore: add reusable tags and externalDocs and move them to the info object

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- add `tags()` and `externalDocs()` methods to the Components models - update also v2 `Components`.
- add `Info`, `License`, `Contact`, `Components` v3 models.
- add `ExternalDocumentations` collection.
- add unit tests. 

cc @smoya 

**Related issue(s)**
https://github.com/asyncapi/spec/pull/792
https://github.com/asyncapi/spec/pull/794

### CODE REVIEW

File: src/models/v2/contact.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@ import { extensions } from './mixins';
 import type { ContactInterface } from '../contact';
 import type { ExtensionsInterface } from '../extensions';
 
-import type { v2 } from '../../spec-types';
+import type { v3 } from '../../spec-types';
 
-export class Contact extends BaseModel<v2.ContactObject> implements ContactInterface {
+export class Contact extends BaseModel<v3.ContactObject> implements ContactInterface {
```
Comment (jonaslagoni) [inline src/models/v2/contact.ts:10]: Is this wrong?

File: src/models/v3/contact.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { BaseModel } from '../base';
+
+import { extensions } from './mixins';
+
+import type { ContactInterface } from '../contact';
+import type { ExtensionsInterface } from '../extensions';
+
+import type { v2 } from '../../spec-types';
+
+export class Contact extends BaseModel<v2.ContactObject> implements ContactInterface {
```
Comment (jonaslagoni) [inline src/models/v3/contact.ts:10]: Should this be v3?

File: src/models/v2/contact.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/models/v2/contact.ts:10]: Yeah. Awesome catch! 👁️ Thanks!

File: src/models/v3/contact.ts (hunk 2 above)
Comment (magicmatatjahu) [reply src/models/v3/contact.ts:10]: Yeah. Awesome catch! 👁️ Thanks!

**Comment by magicmatatjahu** [comment]
@jonaslagoni Done! Could you check again?

**Review by jonaslagoni — APPROVED** [review]
👍

**Pull request merged**